### DISCUSSION TITLE
Possible idea inspiration from checking out other similar (but unique) programs in github?

### DISCUSSION THREAD

**Discussion by Dialogian** [Ross-Patterson/Portals-Desktop-Organization#9] [Ideas]
I found out recently about an alternative folder-portal desktop organization tool, known as Coodesker. I wanted to bring attention to the unique features this program has, to provide possible inspiration for further development & implementation of ideas on this project, especially in case no one here has heard of it yet. 

    There's a lot to check out: It's portable, it can nest portals within each other by enabling tabbed portals, & it has options to auto-sort icons/folder-contents based on custom rules the user can set within the settings. It can even open individual folders within any folder-portal to open up that individual folder's contents within the folder-portal itself, essentially functioning as a basic file-explorer!
    
    Here are some relevant links: https://github.com/coodesker/coodesker-desktop & https://www.coodesker.com/features

**Comment by Ross-Patterson** [comment]
Thanks for this, I had no idea this existed, if I did I would probably have never developed Portals myself!

It looks like a really nice tool, to the point where it actually looks really similar to what I have developed, we obviously had similar ideas of the issues with all of the current tools that exist (Fences etc,).

**Comment by codeisconfusing** [comment]
Please stay away from latest version of "coodesker", it seems to have pushed an update with malware. Version 1.0.3.9 seems safe, but not 1.0.4.0

I've had 1.0.3.9 installed for about 1 month.

Today, I got an upgrade notification, went to install, and Bitdefender went scorched earth on it.

'Advanced Threat Defense

You must restart your device to finalize the cleaning process.
Bitdefender detected potentially malicious behavior and blocked all applications involved.
Detection ID: SuspiciousBehavior.B7B150B04A1374AB'

So I scanned both 1.0.4.0 and 1.0.3.9 installers with some online tools...

Coodesker-x64_1.0.4.0.exe (latest file):

https://opentip.kaspersky.com/077B2F0FEDBB74FA78E5BC24CC9E0DE679388589AEA62A5E2403EA7A94AB98D7/results (shows it as malware / password stealer)

https://www.virustotal.com/gui/file/077b2f0fedbb74fa78e5bc24cc9e0de679388589aea62a5e2403ea7a94ab98d7/detection

https://hybrid-analysis.com/sample/077b2f0fedbb74fa78e5bc24cc9e0de679388589aea62a5e2403ea7a94ab98d7

Coodesker-x64_1.0.3.9.exe (one I had installed):

https://opentip.kaspersky.com/454FD6988590181267FA4490A72047BB0257B3EAA5A9E10B1291DB45B9AE41A4/results (shows as safe)

https://www.virustotal.com/gui/file/454fd6988590181267fa4490a72047bb0257b3eaa5a9e10b1291db45b9ae41a4/detection

https://hybrid-analysis.com/sample/454fd6988590181267fa4490a72047bb0257b3eaa5a9e10b1291db45b9ae41a4

The virus total and hybrid analysis could be false positives for 1.0.3.9 as it installed OK etc, but both Kaspersky and Bitdefender and detecting on 1.0.4.0.

**Comment by PhoenixR1s1ng** [comment]
The multi tabbed box idea is a really good idea. I do believe is could be implemented better. I would much rather the tabs be attached to the header for example

edit - definitely look to see the tabbed feature here. The UI on is a lot better on Portals, but those tabs are pretty nice for sure and I would love to see them in Portals

**Reply by Ross-Patterson** [reply]
This is in the backlog, but requires a bit of work.

Currently Portals was designed with a 1 to 1 relationship of Portal to Folder. I would need to completely overhaul that to have multiple folders to 1 Portal.

Then there is the question:
Should the user add all of the folders they want as tabs
Or
Should it make each subfolder of the parent folder as a tab.

Then there is the customisation element, should each tab be individually customizable, or all the tabs just follow the header styling and each tab content the same content styling.

**Reply by PhoenixR1s1ng** [reply]
Good questions.
While either approach could work for the end user provided they have their folders set up the way they like.

I like the first choice better overall. Personally I like the idea of having 1, 2, 3, or maybe even 4 tabs to allow a portal to share space to allow for easy navigation. But, as is the case of the second option, if there were to be a parent folder with 20+ sub folders then that would probably be too many tabs. So the first option gets my vote.

Functionality is the most important thing here. But, for the sake of having the best tabs I think having customization's like individual color schemes for each tab ect would definitely help it. No one has ever disliked having a customization element after all.

**Comment by bitoolean** [comment]
One reason I'm looking elsewhere is that sometimes when I install or update software (or games), they will create shortcuts on the desktop (again) - what's the purpose of having a desktop organizer if I need to keep managing those (either remember which shortcuts I already have and then deleting the ones the installers will add back to the desktop when updating every time, or when I uninstall software the shortcuts I've moved elsewhere are now invalid). Seems like a way of just linking these icons would make things easier instead of actually moving them (especially since some of them are shared among Windows users). I haven't even tried Portals yet but the fact that it doesn't even mention these things in the features makes me reluctant. Other software partly supports these features it seems and I'll try those first. Although I liked this one better because it looks cleaner.
